astropy/time/core.py

```diff
@@ -1953,14 +1961,13 @@ def __init__(
             from astropy.coordinates import EarthLocation
 
             if isinstance(location, EarthLocation):
-                self.location = location
+                self._location = location
             else:
-                self.location = EarthLocation(*location)
-            if self.location.size == 1:
-                self.location = self.location.squeeze()
-        else:
-            if not hasattr(self, "location"):
-                self.location = None
+                self._location = EarthLocation(*location)
+            if self._location.size == 1:
+                self._location = self._location.squeeze()
+        elif not hasattr(self, "_location"):
```

Comment:
Could define `_location = None` on the `Time` class

Reply:
This works, but I have a personal (slight) aversion for using class attributes as defaults for instance attributes, because I've been bitten by some nasty bugs that would have been easier to figure out if things were kept simpler.
I don't think this matters too much here, so let @taldcroft be the judge, and I'll go with whichever version he prefers :)

Reply:
@neutrinoceros - I suggested something different below, but in general I like class attribute as defaults since they more explicitly highlight the attributes of an object. This is pretty much the entire basis for dataclasses, which I've started using in new code. By using only the defined class attributes the code feels more robust. The days of tossing new attributes onto an object anywhere along the way are gone. :smile:

Reply:
I like dataclasses but they feel entirely different to me (small and self-contained VS large classes + inheritance trees). In general, the issue (to me) is in figuring out where attributes come from. Anyway, we digress ! I like your suggestion best, so let's try that !